internal/controller/tokenprovider/token_provider.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package tokenprovider
+
+import (
+	"context"
+	"time"
+)
+
+type TokenExpiry struct {
+	Token     string
+	ExpiresAt time.Time
+}
+
+type TokenProvider interface {
+	GetToken(ctx context.Context) (TokenExpiry, error)
+}
+
+// MockTokenProvider usd for unit tests to allow pass in token string and expiry.
+type MockTokenProvider struct {
+	Token     string
+	ExpiresAt time.Time
+	Err       error
+}
+
+func (m *MockTokenProvider) GetToken(_ context.Context) (TokenExpiry, error) {
```

Comment:
Always make sure any method implementing an interface to indicate that it does so like elsewhere.
```suggestion
// GetToken implements TokenExpiry.TokenExpiry.
func (m *MockTokenProvider) GetToken(_ context.Context) (TokenExpiry, error) {
```